move/axelar_gateway/sources/auth.move

```diff
@@ -102,13 +95,13 @@ public(package) fun validate_proof(
         EInvalidEpoch,
     );
 
-    let message = MessageToSign {
-        domain_separator: self.domain_separator,
+    let message = proof::new_message_to_sign(
+        self.domain_separator,
         signers_hash,
-        data_hash,
-    };
+        data_hash
+    );
 
-    proof.validate(bcs::to_bytes(&message));
+    proof.validate(message);
```

Comment:
this is a bit weird. MessageToSign is defined in proof, and then returned here but then immediately passed back to the proof module and then encoded. I'd preserve the MessageToSign definition in this module and pass in the bcs encoded message `proof` module shouldn't care about the `MessageToSign` definition. It should be able to validate a proof over an arbitrary message. It's the `auth` module that encodes the message in a certain way